File: rps.py
import random

name = str(input("Enter your name: "))


def welcome():
    kit = ["Rock", "Paper", "Scissor"]

    man_score = 0
    comp_score = 0
    win_score = 5

    game = True

    print("WELCOME TO THE BEST OF THE BEST GAME ")
    print(
        '''
███████████████████████████████████████████████████████████████
██▄─▄▄▀█─▄▄─█─▄▄▄─█▄─█─▄████████▄─▄▄─██▀▄─██▄─▄▄─█▄─▄▄─█▄─▄▄▀██
███─▄─▄█─██─█─███▀██─▄▀██████████─▄▄▄██─▀─███─▄▄▄██─▄█▀██─▄─▄██
██▄▄█▄▄█▄▄▄▄█▄▄▄▄▄█▄▄█▄▄████████▄▄▄███▄▄█▄▄█▄▄▄███▄▄▄▄▄█▄▄█▄▄██
██████████████─▄▄▄▄█─▄▄▄─█▄─▄█─▄▄▄▄█─▄▄▄▄█─▄▄─█▄─▄▄▀███████████
██████████████▄▄▄▄─█─███▀██─██▄▄▄▄─█▄▄▄▄─█─██─██─▄─▄███████████
██████████████▄▄▄▄▄█▄▄▄▄▄█▄▄▄█▄▄▄▄▄█▄▄▄▄▄█▄▄▄▄█▄▄█▄▄███████████
                                             written in python3
                                                coded by hitesh
_______________________________________________________________
            Note: If You choose invalid value, 
                random value will be taken 
‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
        '''
    )

    print(f"winning score: {win_score}")

    while(game):
        print("‾"*62)
        print(
            f" {name}: {man_score}                           Bot: {comp_score}")

        comp = random.randint(0, 2)

        choice12()

        man = int(input(": "))

        if man != 1 or man != 2 or man != 3:
            print(f"{name} has enter invalid value.")
            print("Hence, a random weapon has been taken.\n")
            man = random.randint(1, 3)

        print(f"{name}r Weapon: {kit[man-1]}")
        print(f"Computer Weapon: {kit[comp]}")
        print("\n")

        comp += 1

        comp_score, man_score = AIplayer(comp_score, man_score, man, comp)

        if comp_score == win_score:
            print("Winner is Bot")
            print(f"{name} is loser!")
            print(f"{name} defeated by Bot")

            print("\nWant to play again")
            print("[y/n]:", end=" ")

            inp = str(input())

            if inp == 'n' or inp == 'N':
                game = False
            if inp == 'y' or inp == 'Y':
                man_score = 0
                comp_score = 0

        elif man_score == win_score:
            print(f"{name} Are Winner!")
            print("Hurray")
            print(f"{name} defeated a Bot!")

            print("\nWant to play again")
            print("[y/n]:", end=" ")

            inp = str(input())

            if inp == 'n' or inp == 'N':
                game = False
            if inp == 'y' or inp == 'Y':
                man_score = 0
                comp_score = 0


def choice12():
    print("\n")
    print("Choose your weapon:")
    print("1. Rock")
    print("2. Paper")
    print("3. Scissor")
    print("\n")


def AIplayer(comp_score, man_score, man, comp):
    if comp == 1:   # Rock

        if man == 1:    # Rock
            print("Draw\n")

        elif man == 2:  # Paper
            print(f"{name} win!")
            man_score += 1

        elif man == 3:  # Scissor
            print(f"{name} lose!\n")
            comp_score += 1

    elif comp == 2:  # Paper

        if man == 2:  # Paper
            print("Draw\n")

        elif man == 3:  # Scissor
            print(f"{name} win!")
            man_score += 1

        elif man == 1:  # Rock
            print(f"{name} lose!\n")
            comp_score += 1

    elif comp == 3:  # Scissor

        if man == 3:    # Scissor
            print("Draw\n")

        elif man == 1:  # Rock
            print(f"{name} win!")
            man_score += 1

        elif man == 2:  # Paper
            print(f"{name} lose!\n")
            comp_score += 1
    return comp_score, man_score


welcome()

File: rps.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

int Conclusion(int WEAPON1, int WEAPON2);
int randint(int n);

int main()
{
    //WELCOME TEXT

    printf(
        "███████████████████████████████████████████████████████████████\n"
        "██▄─▄▄▀█─▄▄─█─▄▄▄─█▄─█─▄████████▄─▄▄─██▀▄─██▄─▄▄─█▄─▄▄─█▄─▄▄▀██\n"
        "███─▄─▄█─██─█─███▀██─▄▀██████████─▄▄▄██─▀─███─▄▄▄██─▄█▀██─▄─▄██\n"
        "██▄▄█▄▄█▄▄▄▄█▄▄▄▄▄█▄▄█▄▄████████▄▄▄███▄▄█▄▄█▄▄▄███▄▄▄▄▄█▄▄█▄▄██\n"
        "██████████████─▄▄▄▄█─▄▄▄─█▄─▄█─▄▄▄▄█─▄▄▄▄█─▄▄─█▄─▄▄▀███████████\n"
        "██████████████▄▄▄▄─█─███▀██─██▄▄▄▄─█▄▄▄▄─█─██─██─▄─▄███████████\n"
        "██████████████▄▄▄▄▄█▄▄▄▄▄█▄▄▄█▄▄▄▄▄█▄▄▄▄▄█▄▄▄▄█▄▄█▄▄███████████\n"
        "                                                   written in c\n"
        "                                                      by hitesh\n");

    // Variables
    int win_score = 3;
    int player_score = 0;
    int cpu_score = 0;
    int weapon1, weapon2;
    int results;

    char weapon_list[][7] = {{"Rock"}, {"Paper"}, {"Scissor"}};

    char *player1;
    player1 = (char *)malloc(5 * sizeof(char));

    // Taking player 1 input
    printf("note:- don't use space\n");
    printf("Enter your name: ");
    scanf("%s", player1);

    printf("The game is starting for %s...\n\n", player1);
    printf("Winning score: %d\n‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾\n", win_score);

    // Starting loop

    int game_start = 0;
    int game_end = 3;
    while (game_start < game_end)
    {
        game_start++;

        // If game draws this gives another chance
        if (game_start == 2)
        {
            if (player_score == cpu_score)
            {
                printf("The game is draw, hence one more turn is played.\n");
                game_start = game_start - 1;
            }
        }

        // Displaying scores
        printf("\n%s's score: %d\n", player1, player_score);
        printf("CPU's score: %d\n", cpu_score);
        printf("\n%s' Turn ==>\n\n", player1);

        // weapon selection by player1
        printf("Enter 1 for Rock, 2 for Scissors & 3 for Paper\n");
        printf("> ");
        scanf("%d", &weapon1);
        weapon1 = weapon1 - 1;
        printf("\n%s chose %s\n", player1, weapon_list[weapon1]);

        // weapon selection by cpu
        weapon2 = randint(3);
        printf("CPU chose %s\n\n", weapon_list[weapon2]);

        // Calculating Conclusion
        results = Conclusion(weapon1, weapon2);

        if (results == 1)
        {
            printf("\tDraw\n");
            printf("🏅 %s got 1 point\n", player1);
            printf("🏅 CPU got 1 point\n");
            player_score = player_score + 1;
            cpu_score = cpu_score + 1;
        }
        else if (results == 2)
        {
            printf("🏅 %s got 1 point\n", player1);
            player_score = player_score + 1;
        }
        else if (results == 3)
        {
            printf("🏅 CPU got 1 point\n");
            cpu_score = cpu_score + 1;
        }
    }

    if (player_score > cpu_score)
    {
        printf("\n%s's score: %d\n", player1, player_score);
        printf("CPU's score: %d\n", cpu_score);
        printf("\n🏆 %s is Winner🎉\n", player1);
    }
    else if (player_score < cpu_score)
    {
        printf("\n%s's score: %d\n", player1, player_score);
        printf("CPU's score: %d\n", cpu_score);
        printf("\n🏆 CPU is Winner🎉\n", player1);
    }

    free(player1);

    return 0;
}


int randint(int n)
{
    srand(time(NULL)); // srand takes seed as an input and defined inside stdlib.h
    return rand() % n;
}



int Conclusion(int WEAPON1, int WEAPON2)
{
    int result;
    // for weapon => 0 = rock, 1 = scissors & 2 = paper
    // for result => 1 for draw, 2 = win & 3 = lost

    if (WEAPON1 == 0)
    {
        if (WEAPON2 == 0)
        {
            result = 1;
        }
        else if (WEAPON2 == 1)
        {
            result = 2;
        }
        else if (WEAPON2 == 2)
        {
            result = 3;
        }
    }

    else if (WEAPON1 == 1)
    {
        if (WEAPON2 == 0)
        {
            result = 1;
        }
        else if (WEAPON2 == 1)
        {
            result = 2;
        }
        else if (WEAPON2 == 2)
        {
            result = 3;
        }
    }

    else if (WEAPON1 == 2)
    {
        if (WEAPON2 == 0)
        {
            result = 1;
        }
        else if (WEAPON2 == 1)
        {
            result = 2;
        }
        else if (WEAPON2 == 2)
        {
            result = 3;
        }
    }

    return result;
}
